src/components/Overlay/Overlay.js

```diff
@@ -240,6 +244,9 @@ function Overlay({
             return null;
           },
         }}
+        PaperProps={{
+          ref: (node) => dispatch(setOverlayElement(isMobile ? null : node)),
```

Comment:
To make it more understandable with the useEffect at the beginning do:

Here do: 
ref => (node) => setOverlayNode(node)

then put the logic in the useEffect():
useEffect(() => {
    dispatch(setOverlayElement(isMobile ? null : node))
    return () => dispatch(setOverlayElement(null));
  }, [isMobile, node, dispatch]);

Reply:
Done